### ISSUE TITLE
about training

### ISSUE THREAD

**Issue by traveling121** [CompVis/stable-diffusion#552]
When I run 'main.py', it shows as follows:

/data/stable-diffusion-main/venv/bin/python /data/stable-diffusion-main/main.py 
Global seed set to 23
Traceback (most recent call last):
  File "/data/stable-diffusion-main/main.py", line 535, in <module>
    model = instantiate_from_config(config.model)
  File "/data/stable-diffusion-main/venv/lib/python3.7/site-packages/omegaconf/dictconfig.py", line 356, in __getattr__
    key=key, value=None, cause=e, type_override=ConfigAttributeError
  File "/data/stable-diffusion-main/venv/lib/python3.7/site-packages/omegaconf/base.py", line 237, in _format_and_raise
    type_override=type_override,
  File "/data/stable-diffusion-main/venv/lib/python3.7/site-packages/omegaconf/_utils.py", line 899, in format_and_raise
    _raise(ex, cause)
  File "/data/stable-diffusion-main/venv/lib/python3.7/site-packages/omegaconf/_utils.py", line 797, in _raise
    raise ex.with_traceback(sys.exc_info()[2])  # set env var OC_CAUSE=1 for full trace
  File "/data/stable-diffusion-main/venv/lib/python3.7/site-packages/omegaconf/dictconfig.py", line 352, in __getattr__
    key=key, default_value=_DEFAULT_MARKER_, validate_key=False
  File "/data/stable-diffusion-main/venv/lib/python3.7/site-packages/omegaconf/dictconfig.py", line 443, in _get_impl
    key=key, throw_on_missing_key=True, validate_key=validate_key
  File "/data/stable-diffusion-main/venv/lib/python3.7/site-packages/omegaconf/basecontainer.py", line 78, in _get_child
    throw_on_missing_key=throw_on_missing_key,
  File "/data/stable-diffusion-main/venv/lib/python3.7/site-packages/omegaconf/dictconfig.py", line 480, in _get_node
    raise ConfigKeyError(f"Missing key {key!s}")
omegaconf.errors.ConfigAttributeError: Missing key model
    full_key: model
    object_type=dict

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/stable-diffusion-main/main.py", line 740, in <module>
    if trainer.global_rank == 0:
NameError: name 'trainer' is not defined

Process finished with exit code 1

Why and how to solve it?

**Comment by diamond0910** [comment]
Hi, did you solve it ?

**Comment by LilyDaytoy** [comment]
Hi, did you solve it? I also met this problem

**Comment by ariannaliu** [comment]
Same problem, did you solve that? Thanks!

**Comment by LilyDaytoy** [comment]
I got "name 'trainer' is not defined" error before, this error is because the trainer in the `try` block is not defined yet, but we have already got an exception in some codes before the trainer initialization. In my case, it was because I have some unexpected keys defined in my config file, which are not in the parameter list of the model to be init. 
I tried to debug line by line before the trainer initialization, and found which line the exception showed up.
